Pull in upstream updates and get molecule tests to run and pass again

jsf9k opened the pull request:
## 🗣 Description

In this pull request I pull in upstream updates from [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role) and get the molecule tests to pass again.

I also add support for Kali Linux.

## 💭 Motivation and Context

I need to add this role to the Kali AMI we are building for the COOL.

## 🧪 Testing

All pre-commit hooks and molecule tests pass.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add fedora 29 and 30 to molecule configuration

jsf9k committed:
Trigger LGTM

jsf9k committed:
Merge pull request #18 from cisagov/improvement/updated_skeleton_files_from_latest_molecule

Add updated "molecule init scenario" files

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Merge pull request #20 from cisagov/change_company_name

Change "company" name to the latest trend

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Merge pull request #18 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

dav3r committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Fix action shield link

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

dav3r committed:
Merge pull request #21 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes

felddy committed:
Remove extra line.

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/actions

felddy committed:
Add molecule run

felddy committed:
Remove white space that makes some of the team itchy.

felddy committed:
Merge pull request #22 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Merge pull request #21 from cisagov/pre-commit_autoupdate_2019-11-02

Re-enable ansible-lint and autoupgrade.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Remove vestigial travis file.

felddy committed:
Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Merge branch 'develop' into improvement/action_cache

felddy committed:
Remove search for previous pre-commit caches.
These seem to be very large caches.  Restoring an old one and updating 
it results in a cache larger than the maximum allowed cache size: 200MB. 
 
"Cache size of 254757924 bytes is over the 200MB limit, not saving 
cache."
So if the config changes it is best to just take the cache-miss and 
start from scratch.

felddy committed:
Merge pull request #23 from cisagov/improvement/upstream

Pull in upstream: pre-commit updates, and clean out travis

felddy committed:
Merge pull request #22 from cisagov/improvement/action_cache

Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Add GitHub Action caches for pip and pre-commit.

felddy committed:
Split build workflow into lint and test jobs.
This should speed things up a bit since pre-commit checks are slow, and 
so is the molecule run.

felddy committed:
Merge pull request #25 from cisagov/improvement/action_caches

Add GitHub Action caches for pip and pre-commit, split workflow into two jobs.

felddy committed:
Force pre-commit hooks to use python3. 
Fixes #23

felddy committed:
Autoupdate pre-commit hooks.

felddy committed:
Merge pull request #24 from cisagov/improvement/pre_commit_default_python

Set a default language for pre-commit hooks

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge pull request #26 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Add codeowners file with team OIS maintainers.
See: 
https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

felddy committed:
Merge pull request #25 from cisagov/improvement/codeowners

Add codeowners file with team OIS maintainers.

felddy committed:
Update CODEOWNERS

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge pull request #27 from cisagov/improvement/codeowners

Improvement/codeowners

mcdonnnj committed:
Add seed-isort-config and isort to pre-commit hooks.
Add isort configuration file.

mcdonnnj committed:
Merge pull request #28 from cisagov/improvements/pre-commit_add_isort

Add isort tool to pre-commit hooks.

jsf9k committed:
Make workflow run when a PR is opened, synchronized, or reopened

A user forked cisagov/scan-target-data and created a pull request, but
the required GitHub Action(s) did not run.  This is presumably because
the user does not have Actions enabled in his or her fork.  Ideally,
the required Action(s) would run in cisagov/scan-target-data when a PR
to merge changes back is created.  Based on my reading of this link,
adding the "pull_request" event type should make this happen:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

jsf9k committed:
Improve list formatting

jsf9k committed:
Merge pull request #29 from cisagov/improvement/better-handle-prs-from-forked-repos

Make workflow run when a PR is opened, synchronized, or reopened

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Add .isort.cfg modified by pre-commit hook

jsf9k committed:
Commit changes made by isort pre-commit hook

jsf9k committed:
Merge pull request #28 from cisagov/improvement/upstream-changes

Pull in upstream changes

mcdonnnj committed:
Backported changes to CONTRIBUTING.md from the development guide.

mcdonnnj committed:
Merge pull request #31 from cisagov/improvements/add_wsl_instructions

Cleanup Linux instructions and add WSL references to CONTRIBUTING.md

mcdonnnj committed:
Update Python version used to 3.8
Update actions/checkout to v2
Update formatting to match downstream children

mcdonnnj committed:
Merge pull request #32 from cisagov/improvements/update_python_version_and_checkout_action

Update Python and actions/checkout Versions

mcdonnnj committed:
Run pre-commit autoupdate.

mcdonnnj committed:
Flip cache order to mirror how it is done downstream.

mcdonnnj committed:
Merge pull request #33 from cisagov/improvement/update_pre-commit_hooks

Update pre-commit hooks (and flip Actions cache order in workflow).

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
Update molecule version to the recent preleases for v3.0 as they support recent versions of Ansible. This has also required updating the molecule.yml file to align with 3.0 changes.

mcdonnnj committed:
All references to '-r' for pip calls have been replaced with the more verbose '--requirement'.

mcdonnnj committed:
Adjust spacing for lint directive parameters.

mcdonnnj committed:
Add pip list to see what packages are installed.

mcdonnnj committed:
Pin molecule to <3.
Revert changes to molecule.yml

mcdonnnj committed:
Merge pull request #34 from cisagov/improvements/use_verbose_pip_switches

Use Verbose pip Switches

mcdonnnj committed:
Remove pip list debugging.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
Update comment for molecule version pinning.

mcdonnnj committed:
Merge pull request #29 from cisagov/improvements/pull_in_upstream_changes

Pull in Upstream Changes

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Merge pull request #35 from cisagov/improvement/static_type_checking

Autoupdate pre-commit hooks.  Add mypy.

dav3r committed:
Add Fedora 31
Comment out Fedora 30 and include explanation why

dav3r committed:
Add Fedora 29 and 31.
Add comment explaining why Fedora 30 cannot be used.

dav3r committed:
Sort platforms alphabetically

dav3r committed:
Merge pull request #30 from cisagov/improvement/support_fedora_31

Support Fedora 31

jsf9k committed:
Fix some indenting

This was driving me nuts!

jsf9k committed:
Merge pull request #31 from cisagov/improvement/fix-indenting

Fix some indenting

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-skeleton-updates

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #36 from cisagov/improvement/add-mypy-cache-to-gitignore

Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #32 from cisagov/improvement/upstream-skeleton-updates

Pull in upstream skeleton updates from skeleton-generic

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/switch-to-amazonlinux-2

jsf9k committed:
Switch to Amazon Linux 2

Amazon Linux 1 is ancient stuff at this point.

jsf9k committed:
Merge pull request #33 from cisagov/improvement/switch-to-amazonlinux-2

Switch to amazonlinux 2

mcdonnnj committed:
Incorporate the Python version into keys for pip and pre-commit caches.
This should resolve the issue seen when the Python version changes before there
is an update to .pre-commit-config.yml which results in pre-commit pointing to
a non-existent Python installation.

mcdonnnj committed:
Merge pull request #37 from cisagov/add_python_version_to_cache_keys

Incorporate the Python version into keys for pip and pre-commit caches.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into bugfix/pull_in_pre-commit_cache_fix

mcdonnnj committed:
Merge pull request #34 from cisagov/bugfix/pull_in_pre-commit_cache_fix

Pull in Fix for pre-commit Caching from Upstream

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into improvement/upstream-skeleton-updates

jsf9k committed:
Add missing --requirement include

jsf9k committed:
Add distributions from upstream and modify tasks to work with them all

jsf9k committed:
Update Terraform code for updated molecule-packer-ci-iam-user-tf-module

jsf9k committed:
Prefer single quotes

jsf9k committed:
Comment out Amazon Linux and Fedora

We don't need to support them right now.

jsf9k committed:
Use AWS test credentials

jsf9k committed:
Add AWS_DEFAULT_REGION as env var and specify role to assume

jsf9k committed:
Add packer permissions

Otherwise no role is properly tagged as the build role to assume.

jsf9k commented on terraform/main.tf:
```diff
@@ -1,15 +1,44 @@
 
-  ssm_parameters = ["/vnc/username", "/vnc/password"]
-  user_name      = "test-ansible-role-vnc-server"
+  add_packer_permissions = true
```
We don't need these permissions, but this is the only way to get the proper tagging so we can assume the role in our GitHub Actions workflow.  I don't like this.

The [cisagov/molecule-packer-ci-iam-user-tf-module](https://github.com/cisagov/molecule-packer-ci-iam-user-tf-module) needs to be split up as described in cisagov/molecule-packer-ci-iam-user-tf-module#10.  It serves Packer well, but it longer serves Molecule well at all.

:disappointed:

dav3r replied:
I agree, but I'm thinking that we should remove the SSM parameters from this role and replace them with default values, similar to what I did in https://github.com/cisagov/ansible-role-guacamole/commit/b2a04688df700e394c957087d847731fa7dd39e1.  Then we can pull the SSM parameters when we actually build the AMI, like in https://github.com/cisagov/guacamole-packer/commit/513d9f4f8156c67748d4d7239063bf935ff3b947.

jsf9k replied:
Yes, that is much better! That way the Ansible role is more general. We should in general prefer looking up SSM parameters in Packer or cloud-init instead of Ansible.

jsf9k replied:
Advice taken in commit 7ff65f7.

dav3r reviewed (approved):
Go go gadget updates! 🚁

felddy reviewed (approved):
I won't spoil the broth, but suffice to say this PR is a bursting with bright flavors, some spice, and deep warmth that settles in your belly and satisfies body and soul. 🍅

jsf9k committed:
Prefer defaults to vars

See here for details:
https://codereviewvideos.com/course/ansible-tutorial/video/variable-precedence-where-to-put-your-role-vars

jsf9k committed:
Use defaults instead of SSM parameters

We can override the defaults when running Packer.  This means we no
longer need the Terraform code that sets up the test user.

jsf9k committed:
Remove reference to non-existent requirements.txt

jsf9k committed:
Use vars files with OS-dependent names to set the names of the packages to install

jsf9k committed:
Prefer double quotes where possible

jsf9k commented:
@felddy, @dav3r, and @mcdonnnj - check out commit 884c776.  I think this is a better way to handle package names that are different across distros, even though Kali is a wonky case because `ansible_distribution` and `ansible_os_family` are both poorly named in that case.

It's also [a good use of the `vars` directory](https://codereviewvideos.com/course/ansible-tutorial/video/variable-precedence-where-to-put-your-role-vars).  These things shouldn't need to be overridden, and we don't want them to be overridden accidentally.
